ui-lib/src/main/java/co/electriccoin/zcash/ui/common/datasource/LocalAddressBookDataSource.kt

```diff
@@ -104,45 +119,75 @@ class LocalAddressBookDataSourceImpl(
                             }
                             .toList(),
                 )
-            writeAddressBookToLocalStorage(addressBook!!)
+            writeAddressBookToLocalStorage(addressBook!!, addressBookKey)
             addressBook!!
         }
 
-    override suspend fun deleteContact(addressBookContact: AddressBookContact): AddressBook =
+    override suspend fun deleteContact(
+        addressBookContact: AddressBookContact,
+        addressBookKey: AddressBookKey
+    ): AddressBook =
         withContext(Dispatchers.IO) {
             val lastUpdated = Clock.System.now()
             addressBook =
                 AddressBook(
                     lastUpdated = lastUpdated,
-                    version = 1,
+                    version = ADDRESS_BOOK_SERIALIZATION_V1,
                     contacts =
                         addressBook?.contacts.orEmpty().toMutableList()
                             .apply {
                                 remove(addressBookContact)
                             }
                             .toList(),
                 )
-            writeAddressBookToLocalStorage(addressBook!!)
+            writeAddressBookToLocalStorage(addressBook!!, addressBookKey)
             addressBook!!
         }
 
-    override suspend fun saveContacts(contacts: AddressBook) {
-        writeAddressBookToLocalStorage(contacts)
+    override suspend fun saveContacts(
+        contacts: AddressBook,
+        addressBookKey: AddressBookKey
+    ) {
+        writeAddressBookToLocalStorage(contacts, addressBookKey)
         this@LocalAddressBookDataSourceImpl.addressBook = contacts
     }
 
-    override suspend fun deleteAddressBook() {
-        addressBookStorageProvider.getStorageFile()?.deleteSuspend()
+    override suspend fun resetAddressBook() {
         addressBook = null
     }
 
-    private fun readLocalFileToAddressBook(): AddressBook? {
-        val file = addressBookStorageProvider.getStorageFile() ?: return null
-        return addressBookProvider.readAddressBookFromFile(file)
+    @Suppress("ReturnCount")
+    private suspend fun readLocalFileToAddressBook(addressBookKey: AddressBookKey): AddressBook? {
+        val encryptedFile = addressBookStorageProvider.getStorageFile(addressBookKey)
+        if (encryptedFile != null) {
+            return try {
+                addressBookProvider.readAddressBookFromFile(encryptedFile, addressBookKey)
```

Comment:
After this was written, we changed the design so that it should attempt to delete any unencrypted file after it has successfully read, decrypted, and parsed an encrypted file (for account 0, which is all we currently support). Please make this change.

Reply:
done